src/main/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentSummaryService.scala

```diff
@@ -0,0 +1,99 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.api.business.agents
+
+import cats.data.Validated
+import io.circe.parser.decode
+import io.circe.syntax.EncoderOps
+import uk.gov.nationalarchives.omega.api.business.{ BusinessRequestValidation, BusinessRequestValidationError, BusinessService, BusinessServiceError, BusinessServiceReply, BusinessServiceRequest, InvalidAgentSummaryRequestError }
+import uk.gov.nationalarchives.omega.api.messages.StubData
+import uk.gov.nationalarchives.omega.api.models.ListAgentSummary
+
+import java.text.SimpleDateFormat
+import java.util.Date
+import scala.util.Try
+
+class ListAgentSummaryService(val stubData: StubData) extends BusinessService with BusinessRequestValidation {
+
+  override def process(request: BusinessServiceRequest): Either[BusinessServiceError, BusinessServiceReply] =
+    decode[ListAgentSummary](request.text.getOrElse(defaultRequest)) match {
+      case Left(e) =>
+        Left(AgentSummaryRequestError(s"There was an error reading the message: ${e.getMessage}"))
+      case Right(_) =>
+        Right(
+          ListAgentSummaryReply(
+            stubData
+              .getAgentSummaries()
+              .asJson
+              .toString()
+          )
+        )
+
+    }
+
+  override def validateRequest(request: BusinessServiceRequest): ValidationResult =
+    if (request.text.nonEmpty) {
+      decode[ListAgentSummary](request.text.get) match {
+        case Right(request) =>
+          val versionIdentifiers: List[String] = List("latest", "all")
+          if (request.versionTimestamp.isEmpty || versionIdentifiers.contains(request.versionTimestamp.get))
+            Validated.valid(ListAgentSummaryRequest(None, Some(request)))
+          else {
+            validateDate(request.versionTimestamp.get) match {
+              case Some(_) =>
+                Validated.valid(ListAgentSummaryRequest(None, Some(request)))
```

Comment:
Done

Reply:
For some reason this comment has appeared before the one I was referring to. Instead of passing in None to the `ListAgentSummaryRequest` constructor here you need to pass in the text from the original `BusinessServiceRequest` as it will need to be decoded again in the process method (we will fix this in future refactoring so that we only decode once).

Reply:
Done